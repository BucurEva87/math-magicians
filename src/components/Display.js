import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class Display extends PureComponent {
  render() {
    const { children } = this.props;

    return (
      <div id="result">{children}</div>
    );
  }
}

Display.propTypes = {
  children: PropTypes.string.isRequired,
};
