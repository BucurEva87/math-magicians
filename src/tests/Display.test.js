import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

describe('Claculator interface display', () => {
  test('is render on the screan', () => {
    const { getByTestId } = render(<Display>0</Display>);

    expect(getByTestId('display')).toBeInTheDocument();
  });

  test('is render on the screan', () => {
    const { getByTestId } = render(<Display>0</Display>);

    expect(getByTestId('display')).toHaveTextContent('0');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Display>0</Display>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
