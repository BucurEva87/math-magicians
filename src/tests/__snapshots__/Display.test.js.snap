// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claculator interface display matches snapshot 1`] = `
<div
  data-testid="display"
  id="result"
>
  0
</div>
`;
